src/gui/project/window.cpp

```diff
@@ -276,31 +276,18 @@ namespace Toolbox::UI {
     }
 
     void ProjectViewWindow::onDropEvent(RefPtr<DropEvent> ev) {
-        actionPasteIntoIndex(m_view_index, ev->getMimeData());
-        ev->accept();
-    }
-
-    std::vector<std::string_view> splitLines(std::string_view s) {
-        std::vector<std::string_view> result;
-        size_t last_pos         = 0;
-        size_t next_newline_pos = s.find('\n', 0);
-        while (next_newline_pos != std::string::npos) {
-            if (s[next_newline_pos - 1] == '\r') {
-                result.push_back(s.substr(last_pos, next_newline_pos - last_pos - 1));
-            } else {
-                result.push_back(s.substr(last_pos, next_newline_pos - last_pos));
-            }
-            last_pos         = next_newline_pos + 1;
-            next_newline_pos = s.find('\n', last_pos);
+        MimeData data = ev->getMimeData();
+        auto urls     = data.get_urls();
+        if (!urls) {
+            TOOLBOX_ERROR("Passed mime data did not represent files!");
+            return;
         }
-        if (last_pos < s.size()) {
-            if (s[s.size() - 1] == '\r') {
-                result.push_back(s.substr(last_pos, s.size() - last_pos - 1));
-            } else {
-                result.push_back(s.substr(last_pos));
-            }
+        std::vector<fs_path> paths(urls.value().size());
```

Comment:
Either make this be default constructor and call `paths.reserve(urls.value().size())`, or change the following urls for loop to a C-style `i` based for loop and do `paths[i] = std::move(url);` with `auto &url` as an r-value `auto &&url` instead.

Reply:
We want to do this not only to improve performance by reducing unnecessary allocations or constructor operations of default values (leveraging uninitialized "reserve" memory), but also because the current code according to cpp standard will push back the urls after `urls.value().size()` default inserted elements.

Reply:
Oops yeah when I wrote that I thought that the default constructor reserved elements for some reason.

Reply:
Fixed in [this commit](https://github.com/JoshuaMKW/JuniorsToolbox/pull/36/commits/cac9c3c9bca4409d5fa50b800e99dcf6054c08d1)